### PR TITLE
Poc detox

### PR DESCRIPTION
**Pull request by ggilchrist-ledger**
PR to add initial tests for LLM, add iOS tests into the CI and do a bit of refactoring from @JunichiSugiura 's initial work on Detox. The two tests are for a basic Nano X Onboarding flow and a basic Add Password flow. The CI will then be able to run these tests for iOS as part of the `Detox E2E CI` workflow. 

Not included in PR:
- Android CI tests. Still some issues to iron out on this
- Fast CI tests. They take about 1 hour to run (mostly due to the app building).

https://user-images.githubusercontent.com/87659842/166466076-0ab06796-9264-4ca5-ac0d-01f458c640e3.mov

https://user-images.githubusercontent.com/87659842/166466127-3e76c542-eba0-43e1-b420-02eeadad55e6.mov

### Type

- Development of Mobile E2E Test Framework 
- Addition of iOS CI test workflow for nightly testing

### Context

[LLM automation project](https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/3529048065/PROJECT+Mobile+Test+Framework)

### Parts of the app affected / Test plan

To test:
- Have your local environment setup for Android/iOS React Native development
- It's a good idea to clean and `yarn` the repo first
- Build the app. For example: `yarn detox build --configuration android.debug` or `yarn detox build --configuration ios.staging`
- Run the tests (if you are running a debug build, you will need to have the metro bundler running in another terminal in the directory with `yarn start`): `yarn detox test -c android.debug --loglevel verbose` or `yarn detox test -c ios.staging --loglevel verbose`

### CODE REVIEW

File: e2e/setup.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { execSync } from "child_process";
+import * as bridge from "./bridge/server";
+
+beforeAll(() => {
+  console.log("==============> STARTING BRIDGE INIT");
```
Comment (valpinkman) [inline e2e/setup.js:5]: remove all console.log here

File: e2e/specs/onboarding.spec.js [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
-
-    describe("Import", () => {});
-
-    describe("Restore", () => {});
-
-    describe("Connect", () => {
-      onboard("nanoX", "connect");
-    });
+  beforeAll(async () => {
+    console.log("==============> STARTING ONBOARDING DEVICE LAUNCH");
```
Comment (valpinkman) [inline e2e/specs/onboarding.spec.js:7]: remove all console.logs

File: e2e/specs/onboarding.spec.js [hunk 3]
```diff
@@ -1,29 +1,32 @@
-// @flow
-import { it } from "jest-circus";
-import { cleanLaunch, onboard } from "../engine";
+import OnboardingSteps from "../models/onboarding/onboardingSteps";
+import PortfolioPage from "../models/portfolioPage";
+// import { expectBitmapsToBeEqual } from "../helpers";
```
Comment (valpinkman) [inline e2e/specs/onboarding.spec.js:4]: you can remove unused imports

File: e2e/specs/password.spec.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import GeneralSettingsPage from "../models/settings/generalSettingsPage";
+import PasswordEntryPage from "../models/passwordEntryPage";
+import { delay } from "../helpers";
+import { loadConfig } from "../bridge/server";
+
+const CORRECT_PASSWORD = "passWORD$123!";
+
+describe("Password Lock Screen", () => {
+  beforeAll(async () => {
+    console.log("==============> STARTING PASSWORD DEVICE LAUNCH");
```
Comment (valpinkman) [inline e2e/specs/password.spec.js:12]: remove all console.log

File: e2e/specs/perf-test.spec.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    await retryAction(async () => {
+      const firstAccountButton = element(by.text("Komodo 1"));
+      await waitFor(firstAccountButton).toBeVisible();
+      await firstAccountButton.tap();
+    });
+
+    await device.enableSynchronization();
+
+    // eslint-disable-next-line no-console
```
Comment (valpinkman) [inline e2e/specs/perf-test.spec.js:39]: ok to keep this one ;)

**Review by valpinkman — CHANGES_REQUESTED** [review]
Good overall but some cleanup is needed 👍🏻

File: .github/workflows/detox-ci.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        run: yarn --prefer-offline --frozen-lockfile --network-timeout 100000 --network-concurrency 1
+
+      - name: Install dependencies
+        if: ${{ steps.has-hash-commit-deps.outputs.has-hash-commit-deps == 'false' }}
+        env:
+          JOBS: max
+        run: yarn --prefer-offline --frozen-lockfile --network-timeout 100000    
+
+      - name: Rebuild detox
+        if: ${{ steps.yarn-cache.outputs.cache-hit == 'true' }}
```
[the remark this answers is not in the document]
Comment (ggilchrist-ledger) [reply .github/workflows/detox-ci.yml:72]: The examples I've seen have it as part of the cache hit

File: .github/workflows/detox-ci.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        with:
+          ruby-version: 2.6 # Not needed with a .ruby-version file
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+
+      - name: Install applesimutils
+        run: |
+          brew tap wix/brew
+          brew install applesimutils
+
+      - name: Setup Node
```
[the remark this answers is not in the document]
Comment (ggilchrist-ledger) [reply .github/workflows/detox-ci.yml:36]: Ok. Want me to set it to node 14 or can I continue using the `.nvmrc` file?

File: .github/workflows/detox-ci.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ggilchrist-ledger) [reply .github/workflows/detox-ci.yml:72]: Removed it for now anyway. Can tweak it as we progress and see what's useful or not

**Review by valpinkman — APPROVED** [review]
LGTM :)

File: .github/workflows/detox-ci.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: 14
```
Comment (valpinkman) [inline .github/workflows/detox-ci.yml:39]: `node-version`

**Review by valpinkman — CHANGES_REQUESTED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**